controllers/articles.controllers.js

```diff
@@ -2,15 +2,11 @@ const {selectArticleById, selectArticle} = require("../models/articles.models.js
 
 function getArticlesById (req,res,next){
     const {article_id} = req.params;
-    
-    if (isNaN(article_id)) {
-        return res.status(400).send({ msg: 'Bad request' });
-      }
 
-return selectArticleById(article_id)
+ selectArticleById(article_id)
 .then(article => {
     if (!article) {
-        return res.status(404).send({ msg: 'Article not found' });
+        return Promise.reject({ status: 404, msg: "Article not found" });
```

Comment:
better to create custom errors in your model after you have made the db.query

Also I see you have the promise.reject twice, just once in the model is enough